firedrake/assemble.py

```diff
@@ -602,14 +603,10 @@ def base_form_assembly_visitor(self, expr, tensor, *args):
                     assemble(sub_interp, tensor=tensor.subfunctions[i])
                 return tensor
 
-            # Get the interpolator
-            interp_data = expr.interp_data.copy()
-            default_missing_val = interp_data.pop('default_missing_val', None)
             if rank == 1 and isinstance(tensor, firedrake.Function):
                 V = tensor
-            interpolator = firedrake.Interpolator(expr, V, **interp_data)
-            # Assembly
-            return interpolator.assemble(tensor=tensor, default_missing_val=default_missing_val)
+            interpolator = _get_interpolator(expr, V)
```

Comment:
Any chance we can cache the `interpolator` on the `expr`? We do something similar for the `FormAssembler`

Reply:
This does not need to happen in this PR, but we could start thinking about this.